clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/manifest/KubernetesDeployManifestOperation.java

```diff
@@ -218,6 +158,85 @@ public OperationResult operate(List<OperationResult> _unused) {
     return result;
   }
 
+  private List<KubernetesManifest> bindArtifacts(
+      List<KubernetesManifest> inputManifests, Set<Artifact> boundArtifacts) {
+    List<Artifact> requiredArtifacts = description.getRequiredArtifacts();
+    if (requiredArtifacts == null) {
+      requiredArtifacts = new ArrayList<>();
```

Comment:
Could you move this further down where it is actually used? We could also avoid creating an empty list if all we do is iterate over elements. Same for `optionalArtifacts`